realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -422,7 +422,15 @@ public static AllTypes createOrUpdateUsingJsonObject(Realm realm, JSONObject jso
             }
         }
         if (obj == null) {
-            obj = realm.createObject(AllTypes.class);
+            if (json.has("columnString")) {
+                if (json.isNull("columnString")) {
+                    obj = realm.createObject(AllTypes.class, null);
+                } else {
+                    obj = realm.createObject(AllTypes.class, json.getString("columnString"));
+                }
+            } else {
+                obj = realm.createObject(AllTypes.class);
+            }
         }
         if (json.has("columnString")) {
```

Comment:
It is for objects without primary keys. I could probably be optimised a bit, but I would rather not do it  in this PR.